src/diffusers/loaders/ip_adapter.py

```diff
@@ -141,12 +141,14 @@ def load_ip_adapter(
                     subfolder=os.path.join(subfolder, "image_encoder"),
                 ).to(self.device, dtype=self.dtype)
                 self.image_encoder = image_encoder
+                self.register_to_config(image_encoder=CLIPVisionModelWithProjection)
```

Comment:
`image_encoder` is not supposed to be a configuration variable, though. It's a module. Could I get some explanation here?

Reply:
hi, I have added it to force the internal configuration update, since when you call `pipeline` after `load_ip_adapter`, this component is still `None`:

```
pipeline.load_ip_adapter("h94/IP-Adapter", subfolder="models", weight_name="ip-adapter-full-face_sd15.bin")
pipeline
```

Output:
```
StableDiffusionPipeline {
  "_class_name": "StableDiffusionPipeline",
  "_diffusers_version": "0.25.0.dev0",
  "_name_or_path": "SG161222/Realistic_Vision_V4.0_noVAE",
  "feature_extractor": [
    null,
    null
  ],
  "image_encoder": [
    null,
    null
  ],
  "requires_safety_checker": null,
  "safety_checker": [
    null,
    null
  ],
  "scheduler": [
    "diffusers",
    "DDIMScheduler"
  ],
  "text_encoder": [
    "transformers",
    "CLIPTextModel"
  ],
  "tokenizer": [
    "transformers",
    "CLIPTokenizer"
  ],
  "unet": [
    "diffusers",
    "UNet2DConditionModel"
  ],
  "vae": [
    "diffusers",
    "AutoencoderKL"
  ]
}
```

same for feature extractor

Reply:
I see. @patrickvonplaten WDYT? Would you have suggested a better way to tackle this?

Reply:
I think doing `self.register_to_config(image_encoder=CLIPVisionModelWithProjection)` is exactly right here!